export_database.py

```diff
@@ -0,0 +1,120 @@
+            print("error in inserting: ", ex)
+            defaulters.append(notice)
+
+    if(len(defaulters)):
+        print("Defaulters Present!!!")
+    else:
+        print("Yohoo!! No Defaulters")
+
+
+    print("Saving defaulters and repeated notices...")
+    open("defaulters.bson", "w").write(dumps(defaulters))
+    open("repeated_notices.bson","w").write(dumps(repeated_notices))
+
+    print("Export Complete!")
+    mc_old.close()
+    mc_new.close()
+
+def insert_notice(notice, mc_new):
+    '''
+        Tries to insert a notice in default database of MongoClient passed as second argument
+        Returns 0 if insert successful
+        Returns 2 if entry already in database
+        Returns 1 if some other error occurs
+    '''
+    try:
+        mc_new.get_default_database().notices.insert(notice)
+        print("inserted specific notice: ", notice)
+        return 0
+    except errors.DuplicateKeyError:
+        print("specific entry already in database")
+        return 2
+    except errors.ConnectionFailure:
+        print("Error while connecting to DB")
+        return 1
+    except Exception as ex:
+        print("error in specific inserting: ", ex)
+        return 1
+
+def insert_from_file(filename, further_defaulter_filename = "further_defaulters.bson", further_repeated_filename = "further_repeated.bson"):
+    '''
+        Reads a bson file with name 'filename' and tries to insert each notice in the bson to the target Database
+
+        If a notice fails to be inserted it is pushed to further_defaulters and finally saved in file with filename 'further_defaulter_filename'
+        If a notice already exists in target database, it is pushed to 'further_repeated_filename'
+    '''
+    mc_new = MongoClient(new_mongodb_uri)
+    print("connected to new DB")
+
+    notices = loads(open(filename, "r").read())
+    print("Notice count: {}".format(len(notices)))
+    further_defaulters = []
+    further_repeated = []
+    for notice in notices:
+        out = insert_notice(notice, mc_new)
+        if(out == 1):
+            further_defaulters.append(notice)
+        elif(out == 2):
+            further_repeated.append(notice)
+
+    print("Further defaulters count: {}".format(len(further_defaulters)))
+    print("Further repeated count: {}".format(len(further_repeated)))
+
+    print("Saving further defaulters to {}".format(further_defaulter_filename))
+    open(further_defaulter_filename, "w").write(dumps(further_defaulters))
+
+    print("Saving further repeated to {}".format(further_repeated_filename))
+    open(further_repeated_filename, "w").write(dumps(further_repeated))
+
+    print("Attempt to insert from file: {} complete".format(filename))
+
+def start_database_export():
+    '''
+        Script for exporting database
+        Using OLD_MONGODB_URI in env to act as original database
+        Add NEW_MONGODB_URI in env to act as target database
+    '''
+    export_db()
+    # insert_from_file("defaulters.bson")
```

Comment:
I think you want this to be available in case someone wants to run on defaulters - in that case, do you think adding a command-line argument like, `--try-defaulters` would be better than expecting the developer to uncomment?

If my assumption is true that this is needed for defaulters, that answers the last two questions about functions as well.